t/05-scheduler-dependencies.t

```diff
@@ -385,9 +440,56 @@ $t->get_ok('/api/v1/mm/children/done')->status_is(200)->json_is('/jobs' => [$job
 
 # job_grab now should return jobs from clonned group
 # we already called job_set_done on jobE, so worker 6 is available
-$job = OpenQA::Scheduler::Scheduler::job_grab(workerid => $worker_ids[5], allocate => 1);
-is($job->{id},                  $jobB2, "jobB2");            #lowest prio of jobs without parents
-is($job->{settings}->{NICVLAN}, 2,      "different vlan");
+
+# diag "A2 : $jobA2";
+# diag "B2 : $jobB2";
+# diag "C2 : $jobC2";
+# diag "D2 : $jobD2";
+# diag "E2 : $jobE2";
+# diag "F2 : $jobF2";
+
+# First is going A2 -> D2 -> E2
+schedule();
+ok(exists $sent->{job}->{$jobA2}, " $jobA2 was assigned ") or die diag "A2 $jobA2 wasn't scheduled";
+$job = $sent->{job}->{$jobA2}->{jobhash};
+is($job->{id}, $jobA2, "jobA2");    #lowest prio of jobs without parents
+is($job->{settings}->{NICVLAN}, 2, "different vlan") or die diag explain $job;
+
+# Then B2, but since we took all the 3 left workers for A2, D2, E2, creating another one
+# XXX: Weirdly enough,
+my $new_worker_ids = $c->_register($schema, 'host', "10", \%workercaps);
+$c->_register($schema, 'host', "11", \%workercaps);
+$c->_register($schema, 'host', "12", \%workercaps);
+$c->_register($schema, 'host', "13", \%workercaps);
+$c->_register($schema, 'host', "14", \%workercaps);
+
+schedule();
```

Comment:
Why is schedule called 3 times? I remember that @mudler had to bend a bit the test by adding more workers, but the first one should be enough right?

Reply:
This is d73f4f44a7170fdfc152c4602ae0263e70feb673, but of course schedule does different things if more workers are available.